pkg/docker/config/config.go

```diff
@@ -72,20 +73,25 @@ func SetAuthentication(sys *types.SystemContext, registry, username, password st
 	})
 }
 
-// GetAuthentication returns the registry credentials stored in
-// either auth.json file or .docker/config.json
-// If an entry is not found empty strings are returned for the username and password
-func GetAuthentication(sys *types.SystemContext, registry string) (string, string, error) {
+// GetAuthenticationOAuth2 reads the auth including identity tokens. The
+// previous API GetAuthentication only has support for username and password. To
+// maintain the backward competibility, we keep the old API. To get the support
+// for oauth2 in docker registry authentication, callers should be using this
+// API.
+func GetAuthenticationOAuth2(sys *types.SystemContext, registry string) (types.DockerAuthConfig, error) {
```

Comment:
As mentioned elsewhere, a non-OAuth2-specific name would be better.

The doc string should say what the function does or how to use it, not describe its history (history is best kept in commit messages or PRs). Callers of _this one_ don’t care what “previous API” was or why we keep it. OTOH they _do_ care about what happens when no data is found.